src/components/audio-player/audio-player.js

```diff
@@ -0,0 +1,253 @@
+
+		// Scrubbing
+		this.scrubStartAreaEl.addEventListener("touchstart", this.beginScrubbing, { passive: false });
+		this.scrubStartAreaEl.addEventListener("mousedown", this.beginScrubbing);
+	};
+
+
+	// okay so what's slow is actually waiting for timeupdates, and those being throttled.
+	// timeupdates + lerp = flickering. :\ hmm
+	handleTimeChange = () => {
+		requestAnimationFrame(this._handleTimeChange);
+	};
+
+	_handleTimeChange = () => {
+		const duration = this.audioEl.duration;
+		const elapsed = this.audioEl.currentTime;
+		this.setDisplayTime(elapsed, duration);
+		this.drawProgress(elapsed, duration);
+	}
+
+	handleEnd = () => {
+		this.handleTimeChange();
+		this.setPause();
+	};
+
+	setDisplayTime = (elapsed, duration) => {
+		this.currentTimeEl.innerHTML = timeFormatter(elapsed);
+		this.timeRemainingEl.innerHTML = timeFormatter(duration - elapsed);
+	};
+
+	canUpdateAuotmatically = () => !this.isScrubbing;
+
+	drawProgress = (elapsed, duration, width = 10000) => {
+		this.progressBarCanvas.save();
+		this.progressBarCanvas.clearRect(0, 0, width, 6);
+		this.progressBarCanvas.fillStyle = "transparent";
+		this.progressBarCanvas.fillRect(0, 0, width, 6);
+		this.progressBarCanvas.fillStyle = this.isDarkMode
+			? scssExports.colorWhite
+			: scssExports.colorGrey900;
+		this.progressBarCanvas.fillRect(0, 0, (elapsed / duration) * width, 6);
+		this.progressBarCanvas.restore();
+	};
+
+	beginScrubbing = (e) => {
+		e.preventDefault(); // don't fire redundant mouse event, if this was a touch
+		console.log("begin scrubbing", e.type);
+		this.isScrubbing = true;
+		this.initializeScrubbingListeners();
+		this.scrub(e);
+	};
+
+	//prettier-ignore
+	initializeScrubbingListeners = () => {
+		// touch
+		this.scrubbableAreaEl.addEventListener("touchmove", this.scrub, { passive: false });
+		this.scrubbableAreaEl.addEventListener("touchend", this.endScrubbing, { passive: false });
+		this.scrubbableAreaEl.addEventListener("touchcancel", this.endScrubbing, { passive: false });
+		// mouse
+		this.scrubbableAreaEl.addEventListener("mousemove", this.scrub);
+		this.scrubbableAreaEl.addEventListener("mouseup", this.endScrubbing);
+		this.scrubbableAreaEl.addEventListener("mouseleave", this.endScrubbing);
+	}
+
+	scrub = (e) => {
+		e.preventDefault();
+		console.log("scrub", e.type);
+
+		if (this.isScrubbing) {
+			let canvasRectangle = this.progressBarCanvasEl.getBoundingClientRect();
+			let offsetX;
+
+			if (["touchstart", "touchmove"].includes(e.type)) {
+				offsetX = e.touches[0].clientX - canvasRectangle.left;
+			} else {
+				offsetX = e.offsetX;
+			}
+
+			// this.drawProgress(offsetX, canvasRectangle.width);
+			let currentSecond =
+				(offsetX / canvasRectangle.width) * this.audioEl.duration;
+			this.audioEl.currentTime = currentSecond;
+			// this.setDisplayTime(currentSecond, this.audioEl.duration);
+		}
+	};
+
+	endScrubbing = (e) => {
+		e.preventDefault();
+		console.log("end scrubbing", e.type);
```

Comment:
Sorry, I need to take out this debugging code. I'll also remove the comments for prettier-ignore.